src/sensor/ping.cpp

```diff
@@ -359,25 +359,33 @@ void Ping::flash(const QString& fileUrl, bool sendPingGotoBootloader, int baud,
 
     qCDebug(PING_PROTOCOL_PING) << "Finish connection.";
     // TODO: Move thread delay to something more.. correct.
-    QThread::msleep(1000);
-    link()->finishConnection();
+    auto flashSensor = std::function<void(void)>();
```

Comment:
auto flashSensor = [] { ... } would work the same?

Reply:
Because `finishConnection` needs `flashSensor` and `finishConnection` will be called before `flashSensor`. Doing the declaration of each lambda inside the function will break the readability since `finishConnection`  happens before `flashSensor` but the declaration not.